src/plugins/plugin-http.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var shimmer = require('shimmer');
+var url = require('url');
+var isString = require('is').string;
+var merge = require('lodash.merge');
+var httpAgent = require('_http_agent');
+
+function requestWrap(api, request) {
+  var labels = api.labels;
+  function setTraceHeader(parsedOptions, context) {
+    if (context) {
+      return merge(parsedOptions, {
+        headers: {
+          [TraceLabels.TRACE_CONTEXT_HEADER_NAME]: context
+        }
+      });
+    }
+    return parsedOptions;
+  }
+  function parseRequestOptions(requestOptions) {
+    return isString(requestOptions) ?
+      merge(url.parse(requestOptions), {headers: {}}) :
+      merge({headers: {}}, requestOptions);
+  }
+  function extractUrl(parsedOptions) {
+    var uri = parsedOptions;
+    var agent = parsedOptions._defaultAgent || httpAgent.globalAgent;
+    return isString(uri) ? uri :
+      (parsedOptions.protocol || agent.protocol) + '//' +
+      (parsedOptions.hostname || parsedOptions.host || 'localhost') +
+      ((isString(parsedOptions.port) ? (':' + parsedOptions.port) : '')) +
+      (parsedOptions.path || parseRequestOptions.pathName || '/');
+  }
+  function getSpanName(requestOptions) {
+    if (isString(options)) {
+      options = url.parse(requestOptions);
+    }
+    // c.f. _http_client.js ClientRequest constructor
+    return options.hostname || options.host || 'localhost';
+  }
+  function patchedHTTPRequest(requestOptions, callback, request) {
+    var parsedOptions = parseRequestOptions(requestOptions);
+    var uri = extractUrl(parsedOptions);
+    var requestLifecycleSpan = api.createChildSpan({name: 'http', url: uri})
+      .addLabel(api.labels.HTTP_METHOD_LABEL_KEY, parsedOptions.method)
+      .addLabel(api.labels.HTTP_URL_LABEL_KEY, uri);
+    parsedOptions = setTraceHeader(parsedOptions, transaction.getTraceContext());
+    var req = request.call(request, requestOptions, function (res) {
```

Comment:
I think the first argument here should be either `this` or `null` (in the hook it was `this`).